src/time.cr

```diff
@@ -503,6 +507,41 @@ struct Time
     utc(seconds: seconds, nanoseconds: nanoseconds.to_i)
   end
 
+  # Creates a new `Time` instance that preserves the same local date-time
+  # representation (wall clock) but observed in a different *location*.
+  #
+  # ```
+  # tokyo = Time.now(Time::Location.load("Asia/Tokyo"))
+  # new_york = tokyo.to_local_in(Time::Location.load("America/New_York"))
+  # tokyo.to_s("%Y-%m-%d %H:%M:%S") == new_york.to_s("%Y-%m-%d %H:%M:%S") # => true
+  # ```
+  #
+  # Both instances refer to a different instant in time and they are apart from
+  # each other by the difference in zone offsets.
+  # In contrast, `#in` changes to a different location while referring to the
+  # same instant on the time line, thus changing the local date-time representation.
+  #
+  # ```
+  # tokyo = Time.now(Time::Location.load("Asia/Tokyo"))
+  # new_york = tokyo.to_local_in(Time::Location.load("America/New_York"))
+  # new_york - tokyo              # => (tokyo.offset - new_york.offset).seconds
+  # new_york - tokyo.in(location) # => (tokyo.offset - new_york.offset).seconds
+  # ```
+  #
+  # The result of this method is equivalent to going through the calendrical
+  # constructor like this:
+  #
+  # ```
+  # tokyo = Time.now(Time::Location.load("Asia/Tokyo"))
+  # new_york = Time.new(tokyo.year, tokyo.month, tokyo.day, tokyo.hour, tokyo.minute, tokyo.second, nanosecond: tokyo.nanosecond, location: location)
+  # ```
```

Comment:
This is too abstract. The naming doesn't give much hint at what the method means to achieve, and the documentation doesn't really help until the last example. Maybe using fixed datetimes / string representations, instead of an abstract now and some time operations, and an actual use case would help?

For example, the following would be more helpful, and enough to understand:
> Creates a new `Time` instance with only the location changed. Unlike `#in` that will shift the date time so the new Time instance still refers to the same UTC timestamp, `#to_local_in` keeps the same date and time but for another location.
> 
> For example:
> ```crystal
> halloween = Time.utc(2018, 10, 31, 17, 30, 0)
> tokyo = halloween.to_local_in(Time::Location.load("Asia/Tokyo"))
> new_york = halloween.to_local_in(Time::Location.load("America/New_York"))
> tokyo.to_s    # => 2018-10-31 17:30:00 Asia/Tokyo
> new_york.to_s # => 2018-09-31 17:30:00 America/New_York
> ```